release.nix

```diff
@@ -18,7 +19,7 @@ let
         src = utils.src;
         PKG_CACHE_PATH = utils.pkgCachePath;
         PKG_IGNORE_TAG = 1;
-        GIT_DIR = utils.dotGit;
+        GIT_DIR = gitDir;
```

Comment:
I'm confused about this. Originally @amydevs needed to be able to auto resolve the commit hash.

This was the solution she had.

Now what's the solution? We still need to auto resolve the commit hash if we are doing a nix-build on this repo.

The paramaterisation only applies during the flake state.

Reply:
Running npm run build works, but running nix-build is failing with the following.

```

> polykey-cli@0.1.4 build
> node ./scripts/build.js

Running tsc:
tsc -p ./tsconfig.build.json
node:internal/process/promises:288
            triggerUncaughtException(err, true /* fromPromise */);
            ^

[Error: ENOENT: no such file or directory, open '/build/Polykey-CLI/.git/HEAD'] {
  errno: -2,
  code: 'ENOENT',
  syscall: 'open',
  path: '/build/Polykey-CLI/.git/HEAD'
}

Node.js v18.17.1

ERROR: `npm build` failed

Here are a few things you can try, depending on the error:
1. Make sure your build script (build) exists
  If there is none, set `dontNpmBuild = true`.
2. If the error being thrown is something similar to "error:0308010C:digital envelope routines::unsupported", add `NODE_OPTIONS = "--openssl-legacy-provider"` to your derivation
  See https://github.com/webpack/webpack/issues/14532 for more information.

error: builder for '/nix/store/ml0rvfm81va2r7388marjvxkwkik75p4-polykey-cli-0.1.4-linux-x64.drv' failed with exit code 1;
       last 10 log lines:
       > Node.js v18.17.1
       >
       > ERROR: `npm build` failed
       >
       > Here are a few things you can try, depending on the error:
       > 1. Make sure your build script (build) exists
       >   If there is none, set `dontNpmBuild = true`.
       > 2. If the error being thrown is something similar to "error:0308010C:digital envelope routines::unsupported", add `NODE_OPTIONS = "--openssl-legacy-provider"` to your derivation
       >   See https://github.com/webpack/webpack/issues/14532 for more information.
       >
       For full logs, run 'nix log /nix/store/ml0rvfm81va2r7388marjvxkwkik75p4-polykey-cli-0.1.4-linux-x64.drv'.

```

I'm going to revert the merge until this is resolved.

Reply:
Actually, it's not breaking breaking things right now, so we can leave it until it's fixed. CI still passes AFAIK.

Reply:
Generally speaking you want to minimise any env variables that is needed. And all new env variables must be documented in the `.env.example`.

But also I don't think this makes sense atm. There should be no need for it.

Also the thing so rolled back.